api/node.go

```diff
@@ -58,6 +62,76 @@ func (s *Server) NodeUpdate(rw http.ResponseWriter, req *http.Request) error {
 	}
 
 	unode, err := s.m.UpdateNode(node)
-	apiContext.Write(toNodeResource(unode, nodeIPMap[node.Name]))
+	apiContext.Write(toNodeResource(unode, nodeIPMap[node.Name], apiContext))
+	return nil
+}
+
+func (s *Server) DiskUpdate(rw http.ResponseWriter, req *http.Request) error {
+	var diskUpdate DiskUpdateInput
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&diskUpdate); err != nil {
+		return err
+	}
+
+	id := mux.Vars(req)["name"]
+	node, err := s.m.GetNode(id)
+	if err != nil {
+		return errors.Wrap(err, "fail to get node")
+	}
+
+	nodeIPMap, err := s.m.GetManagerNodeIPMap()
+	if err != nil {
+		return errors.Wrap(err, "fail to get node ip")
+	}
+
+	diskUpdateMap := map[string]types.DiskSpec{}
+	for _, uDisk := range diskUpdate.Disks {
+		diskUpdateMap[uDisk.Path] = uDisk
+	}
+
+	originDisks := node.Spec.Disks
```

Comment:
Map is a variable by reference. So `originDisks` will point to `node.Spec.Disks` then all the change will apply to the object. Don't think that's your intent.